src/network/networkConfig.ts

```diff
@@ -0,0 +1,162 @@
+import { NativeInstanceConfig } from '../nativeInstance';
+
+/**
+ * Available HTTP request types.
+ */
+export enum HttpRequestType {
+  ManifestDash = 'manifest/dash',
+  ManifestHlsMaster = 'manifest/hls/master',
+  ManifestHlsVariant = 'manifest/hls/variant',
+  ManifestSmooth = 'manifest/smooth',
+  MediaProgressive = 'media/progressive',
+  MediaAudio = 'media/audio',
+  MediaVideo = 'media/video',
+  MediaSubtitles = 'media/subtitles',
+  MediaThumbnails = 'media/thumbnails',
+  DrmLicenseFairplay = 'drm/license/fairplay',
+  DrmCertificateFairplay = 'drm/certificate/fairplay',
+  DrmLicenseWidevine = 'drm/license/widevine',
+  KeyHlsAes = 'key/hls/aes',
+  Unknown = 'unknown',
+}
+
+/**
+ * Base64-encoded string representing the HTTP request body.
+ */
+export type HttpRequestBody = string;
+
+/** Represents an HTTP request. */
+export interface HttpRequest {
+  /** The {@link HttpRequestBody} to send to the server. */
+  body?: HttpRequestBody;
+  /**
+   * The HTTP Headers of the request.
+   * Entries are expected to have the HTTP header as the key and its string content as the value.
+   */
+  headers: Record<string, string>;
+  /** The HTTP method of the request. */
+  method: string;
+  /** The URL of the request. */
+  url: string;
+}
+
+/**
+ * Base64-encoded string representing the HTTP response body.
+ */
+export type HttpResponseBody = string;
```

Comment:
Why base64 encoded? 🤔

Reply:
We decided on base64 encoded string as there is no native support for RN to send raw data between TS and native. Introducing this might require bigger changes.
This is something that could be improved later (with breaking changes).